packages/lib/services/plugins/PluginService.ts

```diff
@@ -429,9 +448,35 @@ export default class PluginService extends BaseService {
 		await shim.fsDriver().copy(jplPath, destPath);
 
 		// Now load it from the profile directory
-		const plugin = await this.loadPluginFromPath(destPath);
-		if (!this.plugins_[plugin.id]) this.setPluginAt(plugin.id, plugin);
-		return plugin;
+		if (loadPlugin) {
+			const plugin = await this.loadPluginFromPath(destPath);
+			if (!this.plugins_[plugin.id]) this.setPluginAt(plugin.id, plugin);
+			return plugin;
+		} else { return null; }
+	}
+
+	public async installDefaultPlugins(pathToPlugins: string, pluginSettings: PluginSettings, service: PluginService): Promise<PluginSettings> {
+		if (!Setting.value('firstStart')) return pluginSettings;
+		const defaultPlugins = await shim.fsDriver().readDirStats(pathToPlugins);
+
+		for (const plugin of defaultPlugins) {
+			const defaultPluginPath: string = path.join(`${pathToPlugins}/${plugin.path}`);
+			await service.installPlugin(defaultPluginPath, false);
+
+			pluginSettings = produce(pluginSettings, (draft: PluginSettings) => {
+				draft[filename(plugin.path)] = defaultPluginSetting();
+			});
+		}
+		Setting.setValue('firstStart', 0);
+		return pluginSettings;
+	}
+
+	public setSettingsForDefaultPlugins(initialSettings: InitialSettings) {
+		Object.keys(initialSettings).forEach(pluginId => {
```

Comment:
I don't see checks in here whether the settings are already set or not. It sounds like you'd overwrite user settings on each start.

Reply:
Now, I have added checks here.